frontend/test/lit/test_peephole_optimizations.py

```diff
@@ -474,20 +474,25 @@ def h(x: float):
             qml.Hadamard(wires=0)
             return qml.expval(qml.PauliZ(0))
 
+        # CHECK-LABEL: func.func private @h_cancel_inverses1_0
+        # CHECK: {{%.+}} = quantum.custom "Hadamard"() {{%.+}} : !quantum.bit
+        # CHECK: {{%.+}} = quantum.custom "RX"({{%.+}}) {{%.+}} : !quantum.bit
+        # CHECK-NEXT: {{%.+}} = quantum.custom "Hadamard"() {{%.+}} : !quantum.bit
+
+        # CHECK-LABEL: func.func private @g_0
+        # CHECK: {{%.+}} = quantum.custom "RX"({{%.+}}) {{%.+}} : !quantum.bit
+        # CHECK-NEXT: {{%.+}} = quantum.custom "Hadamard"() {{%.+}} : !quantum.bit
+        # CHECK-NEXT: {{%.+}} = quantum.custom "Hadamard"() {{%.+}} : !quantum.bit
+
+        # CHECK-LABEL: func.func private @f_cancel_inverses0_0
         # CHECK: {{%.+}} = quantum.custom "RX"({{%.+}}) {{%.+}} : !quantum.bit
         # CHECK-NOT: {{%.+}} = quantum.custom "Hadamard"() {{%.+}} : !quantum.bit
         # CHECK-NEXT: {{%.+}} = quantum.namedobs {{%.+}}[ PauliZ] : !quantum.obs
         # CHECK-NOT: {{%.+}} = quantum.custom "Hadamard"() {{%.+}} : !quantum.bit
         _f = f(xx)
 
-        # CHECK: {{%.+}} = quantum.custom "RX"({{%.+}}) {{%.+}} : !quantum.bit
-        # CHECK-NEXT: {{%.+}} = quantum.custom "Hadamard"() {{%.+}} : !quantum.bit
-        # CHECK-NEXT: {{%.+}} = quantum.custom "Hadamard"() {{%.+}} : !quantum.bit
         _g = g(xx)
```

Comment:
> Where do you think the order inversion comes from, is it because of how we set the insertion point or something during the inlining? 🤔

I think it comes from inlining, see my suggestion below

Reply:
This was fixed, so I am resolving.